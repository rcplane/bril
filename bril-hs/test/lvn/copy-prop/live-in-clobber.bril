# Ensure that variables live in to basic blocks are not "clobbered"

@main {
  x: int = const 0;
.lbl:
  # x is live into block
  y: int = id x;
  x: int = const 1;
  print y;
}
