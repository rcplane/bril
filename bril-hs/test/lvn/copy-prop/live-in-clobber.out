@main {
  x: int = const 0;
.lbl:
  y: int = id x;
  x: int = const 1;
  print y;
}
